src/scheduler/pbs_sched.c

```diff
@@ -781,41 +781,16 @@ read_sched_cmd(int sock)
 		 * so try read it in non-blocking mode, but don't
 		 * return any failure if fails to read, as we have
 		 * successfully enqueued first command
+		 *
+		 * and if we get priority command then just ignore it
+		 * since we are not yet in middle of schedule cycle
 		 */
-		cmd_prio = malloc(sizeof(sched_cmd));
+		cmd_prio = new_sched_cmd();
 		if (cmd_prio == NULL)
 			return 1; /* return 1 as we have enqueued first command */
-		cmd_prio->cmd = -1;
-		cmd_prio->jid = NULL;
 
-		get_sched_cmd_noblk(sock, cmd);
-		/*
-		 * we got priority command,
-		 *
-		 * now check if its restart cycle cmd then ignore it
-		 * as we are not yet in cycle
-		 */
-		if (cmd_prio->cmd == SCH_SCHEDULE_RESTART_CYCLE) {
-			if (cmd_prio->jid)
-				free(cmd_prio->jid);
-			free(cmd_prio);
-			/*
-			 * no return here,
-			 * as we have successfully enqueued first command
-			 */
-		} else if (cmd_prio->cmd != -1) {
-			/* not restart cycle cmd, just enqueue it */
-			if (!ds_enqueue(sched_cmds, cmd_prio)) {
-				if (cmd_prio->jid)
-					free(cmd_prio->jid);
-				free(cmd_prio);
-				/*
-				 * no error return here,
-				 * as we have successfully enqueued first command
-				 */
-			}
-			/* no need of return here, fall to last return */
-		}
+		get_sched_cmd_noblk(sock, cmd_prio);
+		free_sched_cmd(cmd_prio);
```

Comment:
How can we be sure this is a high priority command?

Reply:
@bhroam We already read one command from given secondary connection, so if we get 2nd command from same connection (aka same server) then it must be priority command only. As once server send non-priority command, it won't any other command (except priority command) till it receives end cycle notification.

However my earlier code was not dependent on above fact, where it was enqueuing command in sched_cmds if not priority command, but @suresh-thelkar finds that confusing code and gave comment to remove it. ([this](https://github.com/openpbs/openpbs/pull/2000/commits/a612648971d373d7281e90e29ae076b2c2ae4c93#diff-5379b5f9c171cb04e099383cb416bf83L798) is the if/else block I am talking about)

Reply:
It looks like this got resolved now. I saw thumbs up sign from @bhroam. 

But just adding my view why I felt the earlier code i.e. having that if - else if block causes confusing.

As per the current code, server at the most can send two commands i.e. regular sched command followed by a super high priority command before it receives an end of cycle. This is possible, what is not possible is server sending regular command followed by another regular command before it receives an end of cycle and the earlier code with that else if block was doing that and hence requested Hiren to remove that part as this will never happen.